lambdas/call-dcmaw-async-cri/src/test/java/uk/gov/di/ipv/core/calldcmawasynccri/service/DcmawAsyncCriServiceTest.java

```
@@ -0,0 +1,118 @@
+    @Mock private CriApiService mockCriApiService;
+
+    @Mock private IpvSessionService mockIpvSessionService;
+    @Mock private CriOAuthSessionService mockCriOAuthSessionService;
+
+    @InjectMocks private DcmawAsyncCriService dcmawAsyncCriService;
+
+    @BeforeEach
+    void setUp() {
+        ipvSessionItem = new IpvSessionItem();
+        ipvSessionItem.setVot(Vot.P2);
+        ipvSessionItem.setIpvSessionId(SESSION_ID);
+    }
+
+    @Test
+    void testStartDcmawAsyncSession() throws Exception {
+        ClientOAuthSessionItem clientOAuthSessionItem = new ClientOAuthSessionItem();
+        var vcResponse =
+                VerifiableCredentialResponse.builder()
+                        .userId(clientOAuthSessionItem.getUserId())
+                        .verifiableCredentials(
+                                List.of(PASSPORT_NON_DCMAW_SUCCESSFUL_VC.getVcString()))
+                        .credentialStatus(VerifiableCredentialStatus.CREATED)
+                        .build();
+
+        when(mockCriApiService.fetchVerifiableCredential(any(), any(), any(), any()))
+                .thenReturn(vcResponse);
+
+        when(mockConfigService.getOauthCriConfig(any())).thenReturn(criConfig);
+        when(mockConfigService.getCriOAuthClientSecret(any())).thenReturn("clientSecret");
+
+        BearerAccessToken accessToken = new BearerAccessToken("accessToken");
+        when(mockCriApiService.fetchAccessToken(any(), any(), any())).thenReturn(accessToken);
+
+        dcmawAsyncCriService.startDcmawAsyncSession(
+                OAUTH_STATE, clientOAuthSessionItem, ipvSessionItem);
+
+        verify(mockIpvSessionService).updateIpvSession(ipvSessionItem);
+        verify(mockCriOAuthSessionService).persistCriOAuthSession(any(), any(), any(), any());
+        verify(mockCriApiService).fetchVerifiableCredential(any(), any(), any(), any());
+    }
+
+    @Test
+    void testStartDcmawAsyncSessionThrowsCriApiException() throws Exception {
```

Are these exception tests needed? They seem to just be checking that we don't swallow exceptions